apps/ralts/src/cache/is-cache-stale.ts

```diff
@@ -0,0 +1,12 @@
+import redis from "./redis";
+
+export async function isCacheStale() {
+  try {
+    const latestTimestamp = await redis.get("timestamp");
+    const oneDayAgo = Date.now() - 1000 * 60 * 60 * 24;
+    return new Date(latestTimestamp).getMilliseconds() > oneDayAgo;
+  } catch (e) {
```

Comment:
yep like the early return + 
thought for some silly reason that getMillisceconds = getTIme :D

Reply:
Thoughts? We could put 'isTImePassed' in a util package
```suggestion
export async function isCacheStale() {
  try {
    const latestTimestamp = await redis.get('timestamp');
    if (!latestTimestamp) return true;
    return isTimePassed({ date: new Date(latestTimestamp), time: 'oneDay' });
  } catch (e) {
    console.log(e);
    return true;
  }
}

const TIMES = {
  oneDay: 1000 * 60 * 60 * 24,
  oneHour: 1000 * 60 * 60,
  oneMinute: 1000 * 60,
  oneSecond: 1000,
} as const;

export function isTimePassed({
  date,
  time,
}: {
  date: Date;
  time: keyof typeof TIMES;
}) {
  return date.getTime() < Date.now() - TIMES[time];
}
```

Reply:
Sure, why not. :D

I changed the parameter `time` to `timeOffset` since it offsets the current datetime by the input. Also thought the use of an enum would be nice here, so we don't have to use any strings and object value lookup by the offset input :D. What do you think?

```suggestion
enum TIMES {
  oneDay = 1000 * 60 * 60 * 24,
  oneHour = 1000 * 60 * 60,
  oneMinute = 1000 * 60,
  oneSecond = 1000,
}

export async function isCacheStale() {
  try {
    const latestTimestamp = await redis.get('timestamp');
    if (!latestTimestamp) return true;
    return isTimePassed({ date: new Date(latestTimestamp), timeOffset: TIMES.oneDay });
  } catch (e) {
    console.log(e);
    return true;
  }
}

export function isTimePassed({
  date,
  timeOffset,
}: {
  date: Date;
  timeOffset: TIMES;
}) {
  return date.getTime() < Date.now() - timeOffset;
}
```

Reply:
Not the source I was hoping to find but explains it good enough: [Don't use Enums in Typescript](https://dev.to/ivanzm123/dont-use-enums-in-typescript-they-are-very-dangerous-57bh#:~:text=One%20of%20the%20reasons%20why,and%20performance%20of%20the%20app.)
It doesnt really hurt to use an enum here but i am a bit more drawn to the "as const" pattern even though it doesnt look as pretty the first time you look at it :D open for debate - dont want to religous about it either

Reply:
Lol for real TypeScript? 😅 Yeah let's insert the type aswell and use the `as const` solution from you then. Sad to see that enums are so broken haha